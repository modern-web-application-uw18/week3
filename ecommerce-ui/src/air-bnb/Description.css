* {
    box-sizing: border-box;
}

.Description {
    text-transform: uppercase;
    font-weight: bolder;
}

.Home-Title {
    text-transform: none;
    font-weight: bold;
    font-size: 1em;
}