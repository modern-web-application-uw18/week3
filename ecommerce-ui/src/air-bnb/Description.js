import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Location from './Location';
import './Description.css';

class Description extends Component {

    render() {

        return (
            <div className="Description">
                <div>{this.props.houseType} &middot; <Location location={this.props.location}/></div>
                <div className="Home-Title">{this.props.title}</div>
            </div>
        );
    }
}

Description.propTypes = {
    title: PropTypes.string.isRequired,
    houseType: PropTypes.string.isRequired,
    location: PropTypes.instanceOf(Object)
};

Description.defaultProps = {
    title: 'No title',
    houseType: 'House'
};

export default Description;