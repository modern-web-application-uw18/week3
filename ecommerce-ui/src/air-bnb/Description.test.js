import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Description from './Description';

describe('Description', () => {
    it('renders without crashing', () => {
        const location = {city: 'Tokyo', country: 'Japan'};
        const div = document.createElement('div');
        ReactDOM.render(<Description
            title="Upscale apartment for two in Tokyo suburb"
            houseType="Apartment"
            location={location} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render with title, house type, city, country', () => {
        const location = {city: 'Valencia', country: 'Spain'};
        const component = TestRenderer.create(
            <Description title="Urban home for six in Valencia"
                         houseType="Entire House"
                         location={location} />);
            expect(component.toJSON()).toMatchSnapshot();
    });
});