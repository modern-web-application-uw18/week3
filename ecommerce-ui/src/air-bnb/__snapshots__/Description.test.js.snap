// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should render with title, house type, city, country 1`] = `
<div
  className="Description"
>
  <div>
    Entire House
     Â· 
    <span>
      Valencia
      , 
      Spain
    </span>
  </div>
  <div
    className="Home-Title"
  >
    Urban home for six in Valencia
  </div>
</div>
`;
